app-antivirus/clamav/clamav-1.2.0_rc.ebuild

```diff
@@ -0,0 +1,398 @@
+CDEPEND="
+	acct-group/clamav
+	acct-user/clamav
+	app-arch/bzip2
+	dev-libs/json-c:=
+	dev-libs/libltdl
+	dev-libs/libmspack
+	dev-libs/libpcre2:=
+	dev-libs/libxml2
+	dev-libs/openssl:=
+	>=sys-libs/zlib-1.2.2:=
+	virtual/libiconv
+	!libclamav-only? ( net-misc/curl )
+	clamapp? ( sys-libs/ncurses:= net-misc/curl )
+	elibc_musl? ( sys-libs/fts-standalone )
+	jit? ( <sys-devel/llvm-$((${LLVM_MAX_SLOT} + 1)):= )
+	milter? ( mail-filter/libmilter:= )
+	rar? ( app-arch/unrar )
+	test? ( dev-python/pytest )
+"
+
+BDEPEND="
+	virtual/pkgconfig
+	>=virtual/rust-1.61
+	doc? ( app-doc/doxygen )
+	test? (
+		${PYTHON_DEPS}
+		$(python_gen_any_dep 'dev-python/pytest[${PYTHON_USEDEP}]')
+	)
+"
+
+DEPEND="${CDEPEND}
+	test? ( dev-libs/check )"
+
+RDEPEND="${CDEPEND}
+	selinux? ( sec-policy/selinux-clamav )"
+
+python_check_deps() {
+	python_has_version -b "dev-python/pytest[${PYTHON_USEDEP}]"
+}
+
+pkg_setup() {
+	use jit && llvm_pkg_setup
+	use test && python-any-r1_pkg_setup
+}
+
+src_configure() {
+	use elibc_musl && append-ldflags -lfts
+	use ppc64 && append-flags -mminimal-toc
+
+	local mycmakeargs=(
+		-DAPP_CONFIG_DIRECTORY="${EPREFIX}"/etc/clamav
+		-DBYTECODE_RUNTIME=$(usex jit llvm interpreter)
+		-DCLAMAV_GROUP="clamav"
+		-DCLAMAV_USER="clamav"
+		-DDATABASE_DIRECTORY="${EPREFIX}"/var/lib/clamav
+		-DENABLE_APP=$(usex clamapp ON OFF)
+		-DENABLE_CLAMONACC=$(usex clamonacc ON OFF)
+		-DENABLE_DOXYGEN=$(usex doc)
+		-DENABLE_EXPERIMENTAL=$(usex experimental ON OFF)
+		-DENABLE_EXTERNAL_MSPACK=ON
+		-DENABLE_JSON_SHARED=ON
+		-DENABLE_MAN_PAGES=ON
+		-DENABLE_MILTER=$(usex milter ON OFF)
+		-DENABLE_SHARED_LIB=ON
+		-DENABLE_STATIC_LIB=OFF
+		-DENABLE_SYSTEMD=$(usex systemd ON OFF)
+		-DENABLE_TESTS=$(usex test ON OFF)
+		-DENABLE_UNRAR=$(usex rar ON OFF)
+		-DOPTIMIZE=ON
+	)
+
+	if use test ; then
+		# https://bugs.gentoo.org/818673
+		# Used to enable some more tests but doesn't behave well in
+		# sandbox necessarily(?) + needs certain debug symbols present
+		# in e.g. glibc.
+		mycmakeargs+=(
+			-DCMAKE_DISABLE_FIND_PACKAGE_Valgrind=ON
+			-DPYTHON_FIND_VERSION="${EPYTHON#python}"
+		)
+	fi
+
+	if use jit ; then
+		# Suppress CMake warnings that variables aren't consumed if we aren't using LLVM
+		# https://github.com/Cisco-Talos/clamav/blob/main/INSTALL.md#llvm-optional-see-bytecode-runtime-section
+		# https://github.com/Cisco-Talos/clamav/blob/main/INSTALL.md#bytecode-runtime
+		mycmakeargs+=(
+			-DLLVM_ROOT_DIR="$(get_llvm_prefix -d ${LLVM_MAX_SLOT})"
+			-DLLVM_FIND_VERSION="$(best_version sys-devel/llvm:${LLVM_MAX_SLOT} | cut -c 16-)"
+		)
+	fi
+
+	cmake_src_configure
+}
+
+src_install() {
+	cmake_src_install
+	# init scripts
+	newinitd "${FILESDIR}/clamd.initd" clamd
+	newinitd "${FILESDIR}/freshclam.initd" freshclam
+	use clamonacc && \
+		newinitd "${FILESDIR}/clamonacc.initd" clamonacc
+	use milter && \
+		newinitd "${FILESDIR}/clamav-milter.initd" clamav-milter
+
+	rm -rf "${ED}"/var/lib/clamav || die
```

Comment:
Is this still relevant? Why is the `-f` around, isn't this predictable?

Reply:
Pass, this came from the AV teams pre-0.500 ebuilds (but those were updated for CMake builds shortly before I added newer versions). I'll look into it but I'm also loathe to make arbitrary changes to build system stuff that was (theoretically) validated not long before I started doing clamav ebuilds.

Reply:
I think it's there to nuke a bundled virus database, if there still is one:

1. The bundled database conflicts with the one freshclam downloads
2. On a clean install, it prevents users from starting clamav with an outdated virus database (you have to run freshclam first)

I don't think the `-f` is meaningful.

Reply:
Turns out it's not required anyway for modern clamav so dropped.

Reply:
That would also have been me, possibly in combination with the earlier points. We currently run,

```
diropts -o clamav -g clamav
keepdir /var/lib/clamav
```

but maybe that was not working if we allowed clamav's installation routine to create ` /var/lib/clamav` first with some other permissions? Basically you want that directory created once, with owner clamav:clamav, and without having to call chown on anything.

Reply:
Let me confirm those permissions, I suspect you're correct and I've broken them by not deleting the dir first.

I'll change the order if operations around instead and add a comment to be more explicit about why this is happening.